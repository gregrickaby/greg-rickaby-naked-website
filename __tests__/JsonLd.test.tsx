import JsonLD from '@/components/JsonLd'
import '@testing-library/jest-dom'
import {render} from '@testing-library/react'
import {describe, expect, it} from 'vitest'

/**
 * Test suite for the JsonLd component.
 */
describe('JsonLD', () => {
  it('should render', () => {
    const {getByTestId} = render(<JsonLD />)
    const jsonLd = getByTestId('json-ld')
    expect(jsonLd).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const {getByTestId} = render(<JsonLD />)
    const jsonLd = getByTestId('json-ld')
    expect(jsonLd).toMatchSnapshot()
  })
})
