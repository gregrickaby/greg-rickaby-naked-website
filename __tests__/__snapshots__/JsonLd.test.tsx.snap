// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JsonLD > should match snapshot 1`] = `
<script
  data-testid="json-ld"
  type="application/ld+json"
>
  {
  "@context": "https://schema.org",
  "@type": "Person",
  "name": "Greg Rickaby",
  "jobTitle": "Web Developer",
  "url": "https://gregrickaby.com",
  "sameAs": [
    "https://www.instagram.com/gregoryrickaby/",
    "https://www.threads.net/@gregoryrickaby",
    "https://github.com/gregrickaby/",
    "https://linkedin.com/in/gregrickaby/",
    "https://profiles.wordpress.org/gregrickaby/",
    "https://www.amazon.com/author/gregrickaby",
    "https://www.facebook.com/gregrickaby",
    "https://www.goodreads.com/author/show/16999736.Greg_Rickaby",
    "https://codepen.io/gregrickaby",
    "https://www.youtube.com/@GregRickaby",
    "https://gregrickaby.com/feed.xml",
    "https://feedly.com/i/subscription/feed%2Fhttps%3A%2F%2Fgregrickaby.com%2Ffeed.xml"
  ]
}
</script>
`;
